crates/hir_def/src/body.rs

```diff
@@ -363,6 +376,15 @@ impl BodySourceMap {
         self.pat_map.get(&src).cloned()
     }
 
+    pub fn label_syntax(&self, label: LabelId) -> Result<LabelSource, SyntheticSyntax> {
+        self.label_map_back[label].clone()
+    }
```

Comment:
I *think* labels can't be synthetic (ie, we don't produce labels during desugaring), so this should return a value rather than a result. And, more generally, we shouldn't store a `Result` in the source map in the frist palce.

Reply:
Right, I partially based this off of patterns hence why I used this here but I think you are right, we aren't producing new labels.